### PR TITLE
Dịch "softmax-regression_vn" - Phần 6

### LINKED ISSUE #720: Dịch "softmax-regression" - Phần 6
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_linear-networks/softmax-regression_vn.md#L287-L334

### PR DESCRIPTION
**Pull request by antran511**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #720

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [ ] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [ ] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

3. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_linear-networks/softmax-regression_vn.md [hunk 1]
```diff
@@ -290,15 +290,17 @@ We will rely on *likelihood maximization*, the very same concept that we encount
...
+### Log hợp lý (Log-likelihood)
 
 <!--
 The softmax function gives us a vector $\hat{\mathbf{y}}$, which we can interpret as estimated conditional probabilities of each class given the input $x$, 
 e.g., $\hat{y}_1$ = $\hat{P}(y=\mathrm{cat} \mid \mathbf{x})$.
 We can compare the estimates with reality by checking how probable the *actual* classes are according to our model, given the features.
 -->
 
-*dịch đoạn phía trên*
+Hàm softmax đưa cho chúng ta một vector $\hat{\mathbf{y}}$, thứ có thể được diễn giải như là những xác suất có điểu kiện của từng phân loại được ước lượng dựa trên những đặc trưng đã biết.
```
Comment (minhduc0711) [inline chapter_linear-networks/softmax-regression_vn.md]:
```suggestion
Hàm softmax cho chúng ta một vector $\hat{\mathbf{y}}$, có thể được hiểu như các xác suất có điểu kiện của từng lớp biết đầu vào là $x$.
```

File: chapter_linear-networks/softmax-regression_vn.md [hunk 2]
```diff
@@ -290,15 +290,17 @@ We will rely on *likelihood maximization*, the very same concept that we encount
...
+Ví dụ: $\hat{y}_1$ = $\hat{P}(y=\mathrm{cat} \mid \mathbf{x})$.
+Chúng ta có thể so sánh sự ước lượng với thực tế bằng cách kiểm tra độ chắn chắn của những phân loai thực tế dựa trên mô hình của chúng ta với những đặc trưng đã biết.
```
Comment (minhduc0711) [inline chapter_linear-networks/softmax-regression_vn.md]:
```suggestion
Chúng ta có thể so sánh sự ước lượng với thực tế bằng cách kiểm tra độ chắc chắn của mô hình với các lớp *thực sự* khi được cho trước các đặc trưng.
```

File: chapter_linear-networks/softmax-regression_vn.md [hunk 3]
```diff
@@ -312,7 +314,8 @@ Maximizing $P(Y \mid X)$ (and thus equivalently minimizing $-\log P(Y \mid X)$)
 This yields the loss function (we dropped the superscript $(i)$ to avoid notation clutter):
 -->
 
-*dịch đoạn phía trên*
+Cực đại hoá $P(Y \mid X)$ (và vì vậy tương đương với cực tiểu $-\log P(Y \mid X)$) giúp việc dự đoán mẫu chính xác.
```
Comment (minhduc0711) [inline chapter_linear-networks/softmax-regression_vn.md]:
```suggestion
Cực đại hoá $P(Y \mid X)$ (và vì vậy tương đương với cực tiểu hóa $-\log P(Y \mid X)$) giúp việc dự đoán nhãn tốt hơn.
```

File: chapter_linear-networks/softmax-regression_vn.md [hunk 4]
```diff
@@ -312,7 +314,8 @@ Maximizing $P(Y \mid X)$ (and thus equivalently minimizing $-\log P(Y \mid X)$)
...
+Việc này sẽ cho ra được hàm số mất mát (chúng ta bỏ chỉ số trên $(i)$ để tránh sự rối rắm về kí hiệu):
```
Comment (minhduc0711) [inline chapter_linear-networks/softmax-regression_vn.md]:
```suggestion
Điều này dẫn đến hàm mất mát (chúng tôi lược bỏ chỉ số trên $(i)$ để tránh sự rối rắm về kí hiệu):
```

File: chapter_linear-networks/softmax-regression_vn.md [hunk 5]
```diff
@@ -328,8 +331,14 @@ Note that this is often not possible.
 For example, there might be label noise in the dataset (some examples may be mislabeled).
 It may also not be possible when the input features are not sufficiently informative to classify every example perfectly.
 -->
-
-*dịch đoạn phía trên*
+Bởi vì những lí do sẽ được giải thích trong chốc lát, hàm mất mát này thường được gọi là mất mát *cross-entropy*.
```
Comment (minhduc0711) [inline chapter_linear-networks/softmax-regression_vn.md]: `cross entropy` có trong glossary rồi nhé
```suggestion
Bởi vì những lí do sẽ được giải thích trong chốc lát, hàm mất mát này thường được gọi là mất mát *entropy chéo*.
```

File: chapter_linear-networks/softmax-regression_vn.md [hunk 6]
```diff
@@ -328,8 +331,14 @@ Note that this is often not possible.
...
+Ở đây, chúng ta đã sử dụng nó bằng cách xây dựng $\hat{y}$ giống như một phân phối xác suất rời rạc và vector $\mathbf{y}$ là một vector one-hot.
+Vì thế, tổng của tất cả toạ độ $j$ sẽ triệt tiêu ngoại trừ một số hạng.
```
Comment (minhduc0711) [inline chapter_linear-networks/softmax-regression_vn.md]: Nó nhân với 0 mất hết nên nói `triệt tiêu` k đúng lắm
```suggestion
Vì thế, tổng các số hạng với chỉ số $j$ sẽ biến mất ngoại trừ duy nhất một số hạng.
```

File: chapter_linear-networks/softmax-regression_vn.md [hunk 7]
```diff
@@ -328,8 +331,14 @@ Note that this is often not possible.
...
+Bỡi tất cả $\hat{y}_j$ đều là những xác suất, log của chúng không bao giờ lớn hơn $0$.
```
Comment (minhduc0711) [inline chapter_linear-networks/softmax-regression_vn.md]:
```suggestion
Bởi mọi $\hat{y}_j$ đều là xác suất, log của chúng không bao giờ lớn hơn $0$.
```

File: chapter_linear-networks/softmax-regression_vn.md [hunk 8]
```diff
@@ -328,8 +331,14 @@ Note that this is often not possible.
...
+Kết quả là hàm mất mát sẽ không thể giảm thêm được nữa nếu chúng ta dự đoán chính xác $y$ với một *sự chắc chắn*, đó là, nếu $P(y \mid x) = 1$ cho mẫu chính xác.
```
Comment (minhduc0711) [inline chapter_linear-networks/softmax-regression_vn.md]:
```suggestion
Vì vậy, hàm mất mát sẽ không thể giảm thêm được nữa nếu chúng ta dự đoán chính xác $y$ với *độ chắc chắn tuyệt đối*, tức $P(y \mid x) = 1$ cho nhãn đúng.
```

File: chapter_linear-networks/softmax-regression_vn.md [hunk 9]
```diff
@@ -328,8 +331,14 @@ Note that this is often not possible.
...
+Chú ý rằng điều này thường không khả thi.
+Ví dụ, nhãn bị nhiễu sẽ xuất hiện trong tập dữ liệu (một vài mẫu bị dán nhầm nhãn).
+Điều đó cũng thường khó xảy ra khi những đặc trưng đầu vào không đủ thông tin để phân biệt từng mẫu một cách hoàn hảo.
```
Comment (minhduc0711) [inline chapter_linear-networks/softmax-regression_vn.md]:
```suggestion
Điều này cũng khó xảy ra khi những đặc trưng đầu vào không chứa đủ thông tin để phân loại các mẫu một cách hoàn hảo.
```

File: chapter_linear-networks/softmax-regression_vn.md [hunk 10]
```diff
@@ -290,15 +290,17 @@ We will rely on *likelihood maximization*, the very same concept that we encount
 ### Log-Likelihood
 -->
 
-### *dịch tiêu đề phía trên*
+### Log hợp lý (Log-likelihood)
```
Comment (rootonchair) [inline chapter_linear-networks/softmax-regression_vn.md]:
```suggestion
### Hàm log hợp lý (Log-likelihood)
```

File: chapter_linear-networks/softmax-regression_vn.md (hunk 2 above)
Comment (tanduong) [reply chapter_linear-networks/softmax-regression_vn.md]: `sự ước lượng` nghe nó kỳ sao ấy? Đó là đang chỉ quá trình (cách ước lượng) hay kết quả (giá trị ước lượng)?

```suggestion
Để biết các ước lượng có sát với thực tế hay không ta tính _khả năng_ mà nhãn thật sự giống với dự đoán của mô hình của chúng ta khi cho trước các đặc trưng.
```

Comment (minhduc0711) [reply chapter_linear-networks/softmax-regression_vn.md]: yeah nó là `giá trị ước lượng`. Vế sau thì mình cảm giác vẫn hơi lủng củng, mình để xuất như sau
```suggestion
Để biết các ước lượng có sát với thực tế hay không, ta kiểm tra xác suất mà mô hình gán cho lớp *thật sự* biết các đặc trưng.
```

Comment (minhduc0711) [reply chapter_linear-networks/softmax-regression_vn.md]: @tanduong btw bạn có thể để lại email để mình mời vào slack thảo luận không? Mình sẽ xóa comment ngay sau đó.

Comment (tanduong) [reply chapter_linear-networks/softmax-regression_vn.md]: dntan09 at gmail nhe bạn.

Comment (minhduc0711) [reply chapter_linear-networks/softmax-regression_vn.md]: @tanduong oh có ai đó đã mời bạn rồi, bạn thử xem email xem có invite của slack không nhé.

Comment (tanduong) [reply chapter_linear-networks/softmax-regression_vn.md]: Câu gốc là thế này:

```
We can compare the estimates with reality by checking how probable the *actual* classes are according to our model, given the features.
```

`according to our model, given the features.` là một cụm, nghĩa là cho feature và dùng model đưa ra dự đoán.
Còn phần trước đó là nếu ta muốn biết dự đoán có đúng hay không thì phải tính khả năng (mình không chắc có nên dịch chữ probable ra xác suất hay không) dự đoán là đúng. Để biết là đúng thì phải so với nhãn thật.

Mọi người xem có thể viết lại gọn hơn hông nhe :))

Comment (minhduc0711) [reply chapter_linear-networks/softmax-regression_vn.md]: mình thấy dịch vậy là hợp lý, có vẻ tác giả đang ám chỉ xác suất có điều kiện `P(prediction == true class | features)`

Comment (tanduong) [reply chapter_linear-networks/softmax-regression_vn.md]: > có vẻ tác giả đang ám chỉ xác suất có điều kiện `P(prediction == true class | features)`
uh.

File: chapter_linear-networks/softmax-regression_vn.md (hunk 10 above)
Comment (lkhphuc) [reply chapter_linear-networks/softmax-regression_vn.md]: Chỗ này chưa chắc đã là `hàm` mà @rootonchair, có thể là giá trị. Mình thấy để nguyên được rồi.

Comment (lkhphuc) [reply chapter_linear-networks/softmax-regression_vn.md]: Chỗ này chưa chắc đã là hàm mà @rootonchair, có thể là giá trị. Mình thấy để nguyên được rồi.

Comment (tanduong) [reply chapter_linear-networks/softmax-regression_vn.md]: Nó chỉ có thể là hàm thôi mà :)

**Comment by duythanhvn** [comment]
@antran511 Cập nhật PR này giúp mình nhé.

**Comment by antran511** [comment]
> @antran511 Cập nhật PR này giúp mình nhé.

E đã update

File: chapter_linear-networks/softmax-regression_vn.md [hunk 11]
```diff
@@ -328,15 +328,17 @@ We will rely on *likelihood maximization*, the very same concept that we encount
...
```
Comment (duythanhvn) [inline chapter_linear-networks/softmax-regression_vn.md]: Từ này đã bổ sung vào Bảng thuật ngữ, cũng như đã được sử dụng ở một vài đoạn văn trước đó; nên mình bỏ đi hen.
```suggestion
### Log hợp lý
```

File: chapter_linear-networks/softmax-regression_vn.md [hunk 12]
```diff
@@ -366,8 +369,14 @@ Note that this is often not possible.
 For example, there might be label noise in the dataset (some examples may be mislabeled).
 It may also not be possible when the input features are not sufficiently informative to classify every example perfectly.
 -->
-
-*dịch đoạn phía trên*
+Bởi vì những lí do sẽ được giải thích trong chốc lát, hàm mất mát này thường được gọi là mất mát *entropy chéo*.
```
Comment (duythanhvn) [inline chapter_linear-networks/softmax-regression_vn.md]:
```suggestion
Bởi vì những lí do sẽ được giải thích sau, hàm mất mát này thường được gọi là mất mát *entropy chéo*.
```
